plugins/dayspan.config.js

```diff
@@ -0,0 +1,52 @@
+export default {
+
+  methods:
+  {
+      
+    getStyleNow()
+    {
+      const millisPerDay = 86400000;
+      const sixHoursToMillis = 21600000;
+      const multiplicator = 1.125;
+      let now = Math.floor((this.now.asTime().toMilliseconds() - sixHoursToMillis) * multiplicator);
+      let delta = now / millisPerDay;
+      let top = delta * this.dayHeight;
+
+      return {
+        position: 'absolute',
+        left: '0px',
+        right: '-1px',
+        top: (top - 1) + 'px',
+        borderTop: this.getStyleNowBorder()
+      };
+    },
+
+    getStyleTimed(details, calendarEvent)
+    {
+      const sixHoursToMillis = 21600000;
+      const multiplicator = 1.125;
+      const isBefore =  calendarEvent.end.year <= this.now.year && calendarEvent.end.dayOfYear < this.now.dayOfYear;
+      const isToday = calendarEvent.end.year == this.now.year && calendarEvent.end.dayOfYear == this.now.dayOfYear;
+      const isAfterEnd = calendarEvent.time.end.asTime().toMilliseconds() < Math.floor((this.now.asTime().toMilliseconds() - sixHoursToMillis) * multiplicator);
+      let past = isBefore || (isToday && isAfterEnd);
+
+      let cancelled = calendarEvent.cancelled;
+      let bounds = calendarEvent.getTimeBounds( this.dayHeight, 1, this.columnOffset );
+
+      let color = this.getStyleColor( details, calendarEvent );
+      let stateColor = this.getStyleColor( details, calendarEvent, past, cancelled );
+
+      return {
+        top: bounds.top + 'px',
```

Comment:
Wird nicht hier die Position des Events berechnet?

`getTimeBounds` liefert 0.0 für 0 Uhr und 1.0 für 24 Uhr zurück mit dem ersten Argument als Skalierungsfaktor und dem letzten als Abstand (?). https://clickermonkey.github.io/dayspan/docs/classes/calendarevent.html#gettimebounds

Mich stört, dass ein Teil des Hacks im Store ist, weil Skalierung & Offset etwas mit dem Layout zu tun haben und nicht mit API- oder Nutzerdaten.

In dieser Methode kann man vielleicht auch anknüpfen, um überlappende Events anders darzustellen.

Auf jeden Fall kannst du das mergen, falls sich noch mehr Möglichkeiten ergeben ist das nice to have!

Reply:
Ja mit bounds(...) könnte man eventuell den multiplicator aus dem Store eliminieren, jedoch nicht dass man 6 Stunden vom Start des Events abziehen muss.